src/realm/list.cpp

```diff
@@ -678,6 +690,84 @@ void Lst<Mixed>::add_index(Path& path, Index index) const
     path.emplace_back(ndx);
 }
 
+bool Lst<Mixed>::nullify(ObjLink link)
+{
+    size_t ndx = find_first(link);
+    if (ndx != realm::not_found) {
+        if (Replication* repl = Base::get_replication()) {
+            repl->list_erase(*this, ndx); // Throws
+        }
+
+        m_tree->erase(ndx);
+        return true;
+    }
+    else {
+        // There must be a link in a nested collection
+        size_t sz = size();
+        for (size_t ndx = 0; ndx < sz; ndx++) {
+            Mixed val = m_tree->get(ndx);
+            if (val.is_type(type_Dictionary)) {
+                auto dict = get_dictionary(ndx);
+                if (dict->nullify(link)) {
+                    return true;
+                }
+            }
+            if (val.is_type(type_List)) {
+                auto list = get_list(ndx);
+                if (list->nullify(link)) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+bool Lst<Mixed>::replace_link(ObjLink old_link, ObjLink replace_link)
+{
+    size_t ndx = find_first(old_link);
+    if (ndx != realm::not_found) {
+        set(ndx, replace_link);
+        return true;
+    }
+    else {
+        // There must be a link in a nested collection
+        size_t sz = size();
+        for (size_t ndx = 0; ndx < sz; ndx++) {
+            Mixed val = m_tree->get(ndx);
+            if (val.is_type(type_Dictionary)) {
+                auto dict = get_dictionary(ndx);
+                if (dict->replace_link(old_link, replace_link)) {
+                    return true;
+                }
+            }
+            if (val.is_type(type_List)) {
+                auto list = get_list(ndx);
+                if (list->replace_link(old_link, replace_link)) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+void Lst<Mixed>::remove_backlinks(CascadeState& state) const
+{
+    size_t sz = size();
+    for (size_t ndx = 0; ndx < sz; ndx++) {
+        Mixed val = m_tree->get(ndx);
+        if (val.is_type(type_TypedLink)) {
```

Comment:
You might want to handle the list/dictionary cases first so that if we do end up supporting a `List<TypedLink>` at some point, you don't have to reorder this code.

Reply:
I am not sure what you mean here. I think it makes sense to try to handle a direct link first.

Reply:
I thought that a nested collection worked similarly to column keys where the collection type and the value type were bitwise encoded together. But I see that is not the case. Ignore my comment :)